src/main.rs

```diff
@@ -1,40 +1,43 @@
 use anyhow::Result;
-use clap::Parser;
-use fuelup::commands::install;
+use fuelup::download::component;
+use fuelup::{fuelup_cli, proxy_cli};
+use std::panic;
+use std::path::PathBuf;
 
-use fuelup::commands::fuelup::{self_update, FuelupCommand};
-use fuelup::commands::install::InstallCommand;
-
-#[derive(Debug, Parser)]
-#[clap(name = "fuelup", about = "Fuel Toolchain Manager", version)]
-struct Cli {
-    #[clap(subcommand)]
-    command: Commands,
+fn is_supported_component(component: &str) -> bool {
+    ["forc", "fuel-core", "forc-fmt", "forc-lsp", "forc-explore"].contains(&component)
```

Comment:
Hmm, I struggled here a bit because it's not clear if we should call forc-fmt, forc-lsp and forc-explore all components. Is this a square/rectangle thing where all plugins are components but not all components can be plugins?